### PR TITLE
fix: the test doesn't detect display:flex in the media query

### LINKED ISSUE #418: Product Landing Page test unsuccessful when wrapped in media query
#### Issue Description

Layout test number 3 in Product Landing Page ("My product landing page should utilize CSS flexbox at least once.") returns a failure when I have my flexbox display style wrapped in a media query to be applied at a minimum width.
When moving the flexbox styling to a regular class style it is returned as a success.

#### Browser Information

* Browser Name, Version: Chrome v67
* Operating System: macOS v10.13.5
* Mobile, Desktop, or Tablet: Desktop

#### Your Code / Link to Your Pen

 https://codepen.io/MjRodge/pen/GGbJYo/

### PR DESCRIPTION
**Pull request by ValeraS**
#### Pre-Submission Checklist

- [x] Your pull request targets the `master` branch of freeCodeCamp/testable-projects-fcc
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] Your changes have been tested either locally or using a newly created CDN based on your fork's testable-projects-fcc/build/bundle.js file
- [x] Working changes have been added to the build by running `npm run build`

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.

- [x] Closes currently open issue: Closes #418

### CODE REVIEW

**Comment by tbushman** [comment]
@ValeraS Awesome. This will also fix #437 . Grateful for your help! This PR needs to be rebased and then I'll merge. Many thanks!

**Comment by ValeraS** [comment]
@tbushman it's ready

**Comment by tbushman** [comment]
LGTM!

File: src/utils/style-sheet-utils.js [hunk 1]
```diff
@@ -31,7 +31,12 @@ export const allCSSRulesAsArray = function(styleSheets) {
       try {
         if (styleSheet.cssRules) {
           // Convert the list of rules into an array.
-          const rulesAsArray = [].slice.call(styleSheet.cssRules);
+          const rulesAsArray = [].slice.call(styleSheet.cssRules)
+            .filter(rule => (
+              rule.type === CSSRule.STYLE_RULE ||
```
Comment (tchaffee) [inline src/utils/style-sheet-utils.js]: By adding this filter, does the function still do what its name claims it does? The function name is `allCSSRulesAsArray` and the docs at the top say "Given a list of styleSheet-like objects, it returns an array of CSSStyleRule objects".

Comment (ValeraS) [reply src/utils/style-sheet-utils.js]: @tchaffee The function name does not match the description initially, but now it does what the description says more precisely. It returns array of CSSStyleRule objects and CSSMediaRule objects with CSSStyleRule objects inside.

Let's rename it as allStyleRulesAsArray for example.

**Pull request merged**